### DISCUSSION TITLE
[x] Areas for upskilling or training for RSEs

### DISCUSSION THREAD

**Discussion by pr4deepr** [RSEAA/RSEAA.github.io#55] [Discussion topics for #RSEAA2023]
Within RSEs there are folks from a wide variety of backgrounds with varying programming and software development skills.
Although this could be domain specific, are there common areas that need  training or upskilling?

For example, formal training for software development practices?

**Comment by SaranjeetKaur** [comment]
I am working on something similar - it is called `open learning for RSEs` but at the moment it looks like it will cover all Researchers not just RSEs.

At the Collaborations Workshop this year, I have also heard some discussions around RSE curriculum for students

**Comment by SaranjeetKaur** [comment]
Would attending the Carpentries training be useful for RSEs wishing to upskill themselves?

**Comment by dwannetts** [comment]
In my experience, technical skills are fairly easily acquired, and their need tends to be somewhat organic.  

Soft skills are a different matter.  

Presentation, marketing, project management all seem to arise on the job, but it's rare that we (I ....) get the chance to get the chance to focus on any of them before something technical demands attention.

**Reply by pr4deepr** [reply]
Good point about project management and presentation.

**Reply by rowlandm** [reply]
I'm aiming to provide skills workshops for RSE leaders - [Research Software Engineer (RSE) Leadership Training Course Outline](https://figshare.com/articles/presentation/Research_Software_Engineer_RSE_Leadership_Training_Course_Outline/22801898)

**Comment by pr4deepr** [comment]
As there isn't a formal training process, how do we identify areas of improvement?? 
Domain agnostic ones would help. 

Things I can think of: training in software engineering practices: defensive programming, documenting, parallelizing code, how to collaborate & code. 

A very interesting workshop I came across which sort of fits this theme of upskilling:
Advanced Scientific Programming in Python
https://aspp.school/wiki/
Although, I can't make it due it being quite far.

**Reply by SaranjeetKaur** [reply]
Suggestion: Try Coursera
Carpentries: https://carpentries.org/index.html

**Reply by pr4deepr** [reply]
Would be nice to collate a list of recommended lessons or courses. 
Also, I feel instructor-led live lessons are more valuable than self-paced.. It could be a personal opinion..

**Comment by robmoss** [comment]
Across our infectious disease modelling networks, myself and a few other RSEs have been (slowly) developing training materials aimed to ECRs whose research primarily involes writing a lot of code, but who don't have a formal background in software engineering: https://git-is-my-lab-book.net/

**Comment by SaranjeetKaur** [comment]
Some resources:

https://carpentries.org/index.html
 
Pradeep Rajasekhar to Everyone (13 Sep 2023, 11:26)
https://software-carpentry.org/lessons/
 
Ziad Al-Bkhetan to Everyone (13 Sep 2023, 11:37)
https://www.melbournebioinformatics.org.au/training-and-events/
 
You to Everyone (13 Sep 2023, 11:38)
SSI: https://www.software.ac.uk
Research Software Camps: https://www.software.ac.uk/tags/research-software-camps
 
Ziad Al-Bkhetan to Everyone (13 Sep 2023, 11:39)
https://www.biocommons.org.au/webinars-workshops
 
Pradeep Rajasekhar to Everyone (13 Sep 2023, 11:39)
https://aspp.school/wiki/

**Comment by ziadbkh** [comment]
Some training opportunities in Bioinformatics 

https://www.biocommons.org.au/webinars-workshops
https://www.melbournebioinformatics.org.au/training-and-events/

**Comment by rowlandm** [comment]
[Getting started guide for tutorials](https://github.com/Research-Data-Workflows/getting-started-guide/wiki) to learn about other domains.

**Reply by rowlandm** [reply]
[Research Software Engineer (RSE) Leadership Training Course Outline](https://figshare.com/articles/presentation/Research_Software_Engineer_RSE_Leadership_Training_Course_Outline/22801898)

**Comment by nrmay** [comment]
Technical events and materials can be advertised to the wider community through the Digital Research Skills Australasia platform: https://dresa.org.au
https://twitter.com/dresa_news

**Reply by pr4deepr** [reply]
oh wow.. thanks for sharing
